test_opensearch/run_tests.py

```diff
@@ -46,7 +46,8 @@ def fetch_opensearch_repo():
     # no repo
     if not exists(repo_path) or not exists(join(repo_path, ".git")):
         subprocess.check_call(
-            "git clone https://github.com/opensearch-project/opensearch %s" % repo_path,
+            "git clone --branch main --single-branch --depth 1 "
```

Comment:
I think the problem with this approach would be, that once we have multiple versions of OpenSearch server (and clients too), the branch 'main' would not contain the requisite commit. For example, say the 'main' is OpenSearch version 3.x, and the OpenSearch server against which we are testing is on version 1.0, then the build hash returned by server will be a commit from '1.0' branch, and might not be present on the 'main' branch..

Reply:
Hmm, that's interesting.

Do you think removing the `--branch main --single-branch` would work?
```py
            "git clone --depth 1 https://github.com/opensearch-project/opensearch %s" % repo_path,
```

Thanks for the review.

Reply:
That still might not work, as the required commit might not be present on the main branch (branch 'main' will get picked up by default). I suggested a method in an earlier comment, https://github.com/opensearch-project/opensearch-py/pull/53#issuecomment-905407817, can you try that, or something similar?

Reply:
I think I got it right this time.
@rushiagr, can you please check #70 when you have time.